vllm/model_executor/layers/sampler.py

```diff
@@ -395,7 +402,8 @@ def _sample(
         sample_metadata[sampling_type] = (seq_group_ids, seq_groups,
                                           is_prompts, sample_indices)
         if sampling_type == SamplingType.GREEDY:
-            greedy_samples = torch.argmax(logprobs[sample_indices], dim=-1)
+            greedy_samples = torch.argmax(logprobs[sample_indices.long()],
```

Comment:
QQ: Why do we need `.long` here?

Reply:
The sample index need to be `long` type here, otherwise torch would raise a runtime error. In this case, the sample indices are int32. The conversion would not actually happen if it's already `long` type.

Reply:
@liangfu Thanks for the clarification! How is it related to this PR then?